biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -1506,11 +1507,37 @@ public void refresh(RepositoryPlugin repo) {
 	 * @throws Exception
 	 */
 	Repository getResourceRepository() throws Exception {
-		List<Repository> plugins = getPlugins(Repository.class);
-		// replace any WorkspaceRepositoryMarker plugin
-		plugins.removeIf(WorkspaceRepositoryMarker.class::isInstance);
-		plugins.add(new WorkspaceRepositoryDynamic(this));
+		return getResourceRepository(STRATEGY.ALL);
+	}
 
+	/**
+	 * Return an aggregate repository of all the workspace's repository and the
+	 * projects repository. This workspace must be gotten for each operation, it
+	 * might become stale over time.
+	 *
+	 * @param includeWorkspace will ignore the workspace to prevent build loops
+	 * @return an aggregate repository
+	 * @throws Exception
+	 */
+	Repository getResourceRepository(STRATEGY strategy) throws Exception {
+		List<Repository> plugins = new LinkedList<>();
+
+		switch (strategy) {
+			case WORKSPACE :
+				plugins.add(new WorkspaceRepositoryDynamic(this));
+				break;
+			case REPOS :
+				plugins.addAll(getPlugins(Repository.class));
+				// replace any WorkspaceRepositoryMarker plugin
+				plugins.removeIf(WorkspaceRepositoryMarker.class::isInstance);
+				break;
+			default :
```

Comment:
This should be `ALL` so that if we add more enum values later, we get a compile error in the switch statement if we fail to add the case for the new enum value.